src/main/java/org/wildfly/security/auth/client/AuthenticationConfiguration.java

```diff
@@ -735,6 +736,10 @@ public AuthenticationConfiguration useRealm(String realm) {
         return new SetRealmAuthenticationConfiguration(this, realm);
     }
 
+    public AuthenticationConfiguration useBearerTokenCredential(BearerTokenCredential credential) {
```

Comment:
FYI something that is happening in parallel is that these methods should be final to prevent the implementations overriding the building.